### PR TITLE
Daily api tests

### LINKED ISSUE #159: In-system daily API tests
## Details
- Have the system run a test once a day that tries to connect to each of the APIs
- If one of the systems fail, send a message to Paul

### PR DESCRIPTION
**Pull request by WillSS13**
## Related Issues
- Resolves #159 

## Changes
- **TL;DR: Current functionality simply tests that we can connect with our API keys**
- Created a test directory for API specific functions
- Created "runTests.js" which executes all of the tests (will eventually be cron-job)
- Installed cli-color to help make the console execution more readable.
- Installed MailerSend package for specific calls
- Added Twilio & MailerSend API keys to .env to test outside of Knock
- Sends specifically formatted emails to Paul for Twilio and Salesforce Errors
- Send an SMS alert to Paul's personal # when Mailersend API's key fails
- Executes once a day at 8:00 AM

## Demo
Replicated by running `node server/test/runTests.js` in the root directory.

https://github.com/First-Tee-Pittsburgh/notifications/assets/97945187/0ba14475-141b-4631-b299-1a68719faf6d

image

image

image

image

### CODE REVIEW

**Review by Bchenjamin — CHANGES_REQUESTED** [review]

File: server/test/testKnock.js [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+  console.log(knockColor(">> Testing Connection\n"));
+  try {
+    const response = await knock.messages.list();
+    console.log(arrow + "Successfully connected to Knock.");
+    console.log(arrow + "Retrieved latest message ID: " + clc.bold(response.items[0].id));
+  } catch (err) {
+    let errorMessage = (err.message || "Unknown error");
+    let errorDetails = JSON.stringify(err, null, 2);
+    console.error(arrow + "Error: " + clc.redBright(errorMessage));
+    // TODO: SEND SYSTEM ALERT THROUGH MAILERSEND
```
[the remark this answers is not in the document]
Comment (WillSS13) [reply server/test/testKnock.js:41]: No, and it can't call one of the Knock workflows if the Knock API is down. Just haven't gotten around to it.

File: server/test/testSalesforce.js [hunk 2]
```diff
@@ -0,0 +1,196 @@
...
+  try {
+    const result = await conn.sobject("Session_Registration__c")
+      .select(`Id, Name, Status__c, Contact__c, Contact__r.Name, Contact__r.Primary_Contact_s_Email__c, Contact__r.Emergency_Contact_Number__c, Contact__r.Contact_Type__c, Contact__r.Participation_Status__c`)
+      .limit(1)
+      .execute();
+
+    console.log(arrow + "Sample Session Registration Name: ", clc.bold(result[0].Name));
+    console.log(arrow + "Sample Session Registration Contact: ", clc.bold(result[0].Contact__c));
+  } catch (err) {
+    console.error(arrow + "Failed to query Session Registration Table:", clc.redBright(err.errorCode));
```
[the remark this answers is not in the document]
Comment (WillSS13) [reply server/test/testSalesforce.js:121]: It does in the email that gets sent to him. See the first image after the demo video

**Review by Bchenjamin — APPROVED** [review]
